applications/CumulusProver.md

```diff
@@ -0,0 +1,183 @@
+# Cumulus MultiBlock Prover
+
+- **Team Name:** Navatim Labs
+- **Payment Details:**
+  - **DOT**: 14uNH42NjxS3c4VSxVyHjZYea5ZyJEYWYxjRzmxKQB75pdRB
+  - **Payment**: 14uNH42NjxS3c4VSxVyHjZYea5ZyJEYWYxjRzmxKQB75pdRB (USDT)
+- **Level:** 2
+
+## 2. Project Overview
+
+The **Cumulus Prover** project aims to integrate a cryptographic proving system into Substrate parachains by extending the Cumulus node architecture. Our approach introduces out-of-the-box support for generating and verifying execution proofs in parachain consensus. This system is composed of two components:
+
+* **Prover Node**: A modified cumulus node responsible for generating cryptographic proofs of block correctness during execution.
+* **Verifier Module**: Integrated with para-validators, enabling them to cheaply verify the correctness of blocks. Since only 5 validators are assigned per `paraId`, this system ensures low overhead.
+
+This architecture introduces a novel and lightweight form of block validation that maintains integrity without compromising performance. This project gives new parachains the ability to opt into a new approach to securing their parachain using a cryptographic proving system while maintaining economic security on Polkadot.
+
+## 3. Project Details
+
+We will modify the cumulus node and the `parachain-system` pallet to implement the proving and verifying logic.
+
+### Technical Implementation
+
+* **Node-Level Modifications (Cumulus)**:
+  **Block Builder API**: Will include new cryptographic data fields into the block header structure. Specific changes include:
+
+```rust
+struct ExtendedHeader {
+    proofs: Vec<Vec<u8>>,
+    proof_root: [u8; 32],
+    execution_cycles: u64,
```

Comment:
I don't understand why you need to change the block header. This could be included as an extra transaction or as a header digest. I don't think you need to change the block header structure for this.